pkg/token/token.go

```diff
@@ -125,15 +140,15 @@ type getCallerIdentityWrapper struct {
 // Generator provides new tokens for the heptio authenticator.
 type Generator interface {
 	// Get a token using credentials in the default credentials chain.
-	Get(string) (string, error)
+	Get(string) (*Token, error)
```

Comment:
I would prefer returning an empty struct to nil since we don't use it as a pointer (modify contents etc.)